src/Synth/ADnote.cpp

```diff
@@ -367,9 +368,10 @@ ADnote::ADnote(const ADnote &orig, ADnote *parent, float *parentFMmod_) :
         if (orig.subFMVoice[i] != NULL)
         {
             subFMVoice[i] = new ADnote*[orig.unison_size[i]];
+            ADnote *parentVoice = (origVoice != NULL) ? origVoice : this;
             for (int k = 0; k < orig.unison_size[i]; ++k)
             {
-                subFMVoice[i][k] = new ADnote(*orig.subVoice[i][k], this, parentFMmod);
+                subFMVoice[i][k] = new ADnote(*orig.subVoice[i][k], parentVoice, parentFMmod);
```

Comment:
Did you check that this actually needed to be fixed? I'm pretty sure I already took this into account.

Near the start of the copy constructor, `origVoice` is set to the parent's `origVoice` if a parent is specified and that parent is not the root:
https://github.com/Yoshimi/yoshimi/blob/56fe5f1db45149acdef61a06a3eebb40957f6892/src/Synth/ADnote.cpp#L155-L156

I probably should've written a comment mentioning this.

Your way might actually be better. I wrote it the way I did because I wanted the copy constructor to only need the immediate parent to be specified, but in hindsight it isn't any worse to have to specify `origVoice` directly, especially since calling the copy constructor on subvoices currently only occurs inside the copy constructor itself.

I suggest moving `parentVoice` to a single assignment before these two if blocks and renaming the `parent` arg to `origVoice_` or similar for consistency, then removing the conditional assignment on lines 155-156 that I quoted above.

In fact, thinking about this makes me wonder if it would cause any problems for `origVoice` to never be null, instead being equal to `this` for the root voice. This should make every `origVoice != NULL` branch unnecessary.

Reply:
You're right, that's much better. Thanks for cleaning that up! Much of my own confusion in this area stems from the use of the words "orig" to mean two different things. I have launched #90 to address this.